metron-platform/metron-common/src/main/java/org/apache/metron/common/dsl/functions/MaaSFunctions.java

```diff
@@ -234,25 +248,36 @@ public Object apply(List<Object> args, Context context) throws ParseException {
       }
       return ret;
     }
+
     @Override
-    public void initialize(Context context) {
-      Optional<Object> clientOptional = context.getCapability(Context.Capabilities.ZOOKEEPER_CLIENT);
-      CuratorFramework client = null;
-      if(clientOptional.isPresent() && clientOptional.get() instanceof CuratorFramework) {
-        client = (CuratorFramework)clientOptional.get();
-      }
-      else {
-        return;
-      }
+    public synchronized void initialize(Context context) {
       try {
-        MaaSConfig config = ConfigUtil.INSTANCE.read(client, "/metron/maas/config", new MaaSConfig(), MaaSConfig.class);
-        discoverer = new ServiceDiscoverer(client, config.getServiceRoot());
-        discoverer.start();
-        context.addCapability(Context.Capabilities.SERVICE_DISCOVERER, () -> discoverer);
-      } catch (Exception e) {
-        LOG.error(e.getMessage(), e);
-        return;
+        Optional<Object> clientOptional = context.getCapability(Context.Capabilities.ZOOKEEPER_CLIENT);
+        CuratorFramework client = null;
+        if (clientOptional.isPresent() && clientOptional.get() instanceof CuratorFramework) {
+          client = (CuratorFramework) clientOptional.get();
+        } else {
+          return;
+        }
+        try {
+          MaaSConfig config = ConfigUtil.INSTANCE.read(client, "/metron/maas/config", new MaaSConfig(), MaaSConfig.class);
+          discoverer = new ServiceDiscoverer(client, config.getServiceRoot());
+          discoverer.start();
+          context.addCapability(Context.Capabilities.SERVICE_DISCOVERER, () -> discoverer);
+          isValidState = true;
+        } catch (Exception e) {
+          LOG.error(e.getMessage(), e);
```

Comment:
If initialization fails, we're just logging the error.  Does it not make sense to fail fast or make some noise?